components/brave_extension/extension/brave_extension/manifest.json

```diff
@@ -69,9 +84,11 @@
     "webRequest",
     "*://*/*",
     "chrome://favicon/*",
+    "webRequestBlocking",
```

Comment:
what is it used for?

Reply:
This is used by Web Discovery Project for:
1. Abort double-fetch of resources which are too big (based on `content-length` header).
2. Remove sensitive headers such as `cookie`, `authorization`, etc. from out-going double-fetch requests.

Reply:
I glanced at the code, and it appears to me that not everything works as intended. E.g. no `onBeforeSendHeaders` listener adds `extraHeader` param [1] that is needed to observe sensitive headers such as `Cookie:` or `Origin:`, thus making the filtering code ineffective. `Authorization:` is not provided to `onBeforeSendHeaders` at all.

The same doc says that modification of `Origin:` might not work well.

[1] https://developer.chrome.com/docs/extensions/reference/webRequest/#life-cycle-of-requests

So I suggest we pay more attention to new permissions and extension APIs we use. Maybe I'm just looking at something outdated?

Reply:
It works, but is a bit less obvious than meets the eye maybe.

1. Webrequest handler registration is done by calling this `addPipelineStep` action: https://github.com/brave/web-discovery-project/blob/main/modules/web-discovery-project/sources/doublefetch-handler.es#L501
2. This then ends up here: https://github.com/brave/web-discovery-project/blob/main/modules/webrequest-pipeline/sources/background.es#L192
3. The code doing the actual registration of the listener then adds the extraHeader param here: https://github.com/brave/web-discovery-project/blob/main/modules/webrequest-pipeline/sources/background.es#L142

It's a bit more convoluted than required because of historical reasons and we could probably simplify it a bit now. This code used to be part of a bigger project, hence some of the designs might appear overkill, but they work. The simplification will happen over time, lots of code has already been removed, simplified, improved over the last few weeks; this will continue.